src/v4/types/observablearray.ts

```diff
@@ -631,28 +629,52 @@ addHiddenProp(ObservableArray.prototype, toStringTagSymbol(), "Array")
 /**
  * Wrap function from prototype
  */
+;["indexOf", "join", "lastIndexOf", "slice", "toString", "toLocaleString"].forEach(funcName => {
+    const baseFunc = Array.prototype[funcName]
+    invariant(
+        typeof baseFunc === "function",
+        `Base function not defined on Array prototype: '${funcName}'`
+    )
+    addHiddenProp(ObservableArray.prototype, funcName, function() {
+        return baseFunc.apply(this.peek(), arguments)
```

Comment:
I'd moved the invariant here indeed, so that people that never call the method aren't killed by it